sys/include/matstat.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2018 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_matstat Matstat - Integer mathematical statistics library
+ * @ingroup     sys
+ * @brief       Library for computing 1-pass statistics
+ *
+ * The Matstat library uses single pass algorithms to compute statistic measures
+ * such as mean and variance over many values. The values can be immediately
+ * discarded after processing, keeping the memory requirement constant
+ * regardless of how many values need to be processed.
+ *
+ * The design goal is to provide basic mathematical statistics operations on
+ * constrained devices with a "good enough" accuracy to be able to provide some
+ * descriptive measures of data. For more accurate measures of statistics, use a
+ * fancier library, or copy the data to a PC.
+ *
+ * It is important to know that using integer operations will result in lower
+ * precision in the computed measures because of truncation.
+ *
+ * @{
+ * @file
+ * @brief       Matstat library declarations
+ *
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ */
+
+#ifndef MATSTAT_H
+#define MATSTAT_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Internal state for computing running statistics
+ */
+typedef struct {
+    int64_t sum;        /**< Sum of values added */
+    uint64_t sum_sq;    /**< Sum of squared values added */
```

Comment:
Same as in the functions, I would prefer `sum_sq` defined close to `offset` as they are used together.